projects/core/src/lib/layer-collections/layer-collection.service.ts

```diff
@@ -45,4 +76,186 @@ export class LayerCollectionService {
+        } else {
+            vectorSymbology = this.createVectorSymbology(resultDescriptor.dataType);
+        }
+
+        const layer = new VectorLayer({
+            name: layerName,
+            workflowId,
+            isVisible: true,
+            isLegendVisible: false,
+            symbology: vectorSymbology,
+        });
+
+        this.projectService.addLayer(layer);
+    }
+
+    private createVectorSymbology(dataType: 'Data' | 'MultiPoint' | 'MultiLineString' | 'MultiPolygon'): VectorSymbology {
+        switch (dataType) {
+            case 'Data':
+                // TODO: cope with that
+                throw Error('we cannot add data layers here, yet');
+            case 'MultiPoint':
+                return ClusteredPointSymbology.fromPointSymbologyDict({
+                    type: 'point',
+                    radius: {type: 'static', value: PointSymbology.DEFAULT_POINT_RADIUS},
+                    stroke: {
+                        width: {type: 'static', value: 1},
+                        color: {type: 'static', color: [0, 0, 0, 255]},
+                    },
+                    fillColor: {type: 'static', color: colorToDict(this.randomColorService.getRandomColorRgba())},
+                });
+            case 'MultiLineString':
+                return LineSymbology.fromLineSymbologyDict({
+                    type: 'line',
+                    stroke: {
+                        width: {type: 'static', value: 1},
+                        color: {type: 'static', color: [0, 0, 0, 255]},
+                    },
+                });
+            case 'MultiPolygon':
+                return PolygonSymbology.fromPolygonSymbologyDict({
+                    type: 'polygon',
+                    stroke: {
+                        width: {type: 'static', value: 1},
+                        color: {type: 'static', color: [0, 0, 0, 255]},
+                    },
+                    fillColor: {type: 'static', color: colorToDict(this.randomColorService.getRandomColorRgba())},
+                });
+        }
+    }
+
+    private addRasterLayer(
+        layerName: string,
+        workflowId: UUID,
+        _resultDescriptor: RasterResultDescriptorDict,
+        symbology?: SymbologyDict,
+    ): void {
+        let rasterSymbologyDict: RasterSymbologyDict;
+        if (symbology && symbology.type === 'raster') {
+            rasterSymbologyDict = symbology as RasterSymbologyDict;
+        } else {
+            rasterSymbologyDict = {
+                type: 'raster',
+                opacity: 1.0,
+                colorizer: {
+                    type: 'linearGradient',
+                    breakpoints: [
+                        {value: 1, color: [0, 0, 0, 255]},
+                        {value: 255, color: [255, 255, 255, 255]},
+                    ],
+                    defaultColor: [0, 0, 0, 0],
+                    noDataColor: [0, 0, 0, 0],
+                },
+            };
+        }
```

Comment:
Is there a chance to generate a palette for a classification?

Reply:
We could generate one, if the measurement is available and is a classification. But to be fair, the default gradient also doesn't really make sense with a range from 1 to 255...